worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -853,33 +857,43 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
   }
 
   def updateDiskInfos(): Unit = this.synchronized {
-    disksSnapshot().filter(_.status != DiskStatus.IO_HANG).foreach { diskInfo =>
-      val totalUsage = diskInfo.dirs.map { dir =>
-        val writers = workingDirWriters.get(dir)
-        if (writers != null) {
-          writers.synchronized {
-            writers.values.asScala.map(_.getDiskFileInfo.getFileLength).sum
+    disksSnapshot()
+      .filter(diskInfo =>
+        diskInfo.status != DiskStatus.IO_HANG || diskInfo.status != DiskStatus.READ_OR_WRITE_FAILURE)
```

Comment:
Do we need to filter `CRITICAL_ERROR` as well?

Reply:
IMO, this only filters the non-critical error status of disk. cc @FMX.

Reply:
Yes. Indeed. This filter is intended to filter the disk that can not response.
If a disk is hung, it can not response.
If a disk has read writer failures, it may not be able to response.